### ISSUE TITLE
Saved resources will preserve default exported variable values

### ISSUE THREAD

**Issue by stryker313** [godotengine/godot#80894]
### Godot version

3.6 beta 2

### System information

Windows 10 gtx 1060 6gb intel i5 4690k

### Issue description

When you save a custom resource with exported vars, it will write the default value to the .tres file

![Screenshot 2023-08-22 09-40-31](https://github.com/godotengine/godot/assets/83078010/6d6295d9-686b-4342-bac6-6713d26f7031)

Comparing to the Animation class
![Screenshot 2023-08-22 09-45-46](https://github.com/godotengine/godot/assets/83078010/f7c4045f-0a69-4145-bbf6-dbb4580e965f)

Only when you set a new value is that value written to file...

This also happens when you extend a Godot class, ie extends Animation, all new properties will have their default values written, but all the original ones belonging to the parent class will still function normally...

This adds a size + loading penalty for adding export vars.

### Steps to reproduce

1. Create a new custom resource
2. Give the custom resource several exported vars
3. Change none of their values
4. Give a class name to the resource
5. Save the resource using ResourceSaver.save(), and save as a .tres
6. Open the file in a text editor, and you will see that the default values are written to file.

### Minimal reproduction project

[PropertySize.zip](https://github.com/godotengine/godot/files/12409324/PropertySize.zip)

**Comment by dalexeev** [comment]
MRP for 4.x: [PropertySize.zip](https://github.com/godotengine/godot/files/12409928/PropertySize.zip).

**Comment by Calinou** [comment]
We may want to add an annotation to control this behavior, as it can be desired in some cases. This needs more core changes though, so it can be left for later.

**Comment by dalexeev** [comment]
> We may want to add an annotation to control this behavior (with the default being "don't save"), as it can be desired in some cases.

Yes, I suggested something like that. For example, theme property overrides have a checkbox - if you check it, the value will be saved always. And also when inheriting scenes you can "pin" property.

But I think this applies to all properties, both custom and standard. In my opinion this is definitely a bug, because it is inconsistent with the standard resource properties and any node properties: they are not saved in the file if they are default.

**Comment by stryker313** [comment]
> #80897 
4.x only?

**Comment by dalexeev** [comment]
> 4.x only?

PRs should first be opened against the master branch (if the bug is in master). Then the fix can be cherry-picked or reimplemented for older supported branches, if it's considered desirable and safe enough.
